### PR TITLE
feat(opa): Adding jq OPA builtin for selection

### PR DESCRIPTION
**Pull request by elizabethhealy**
Add an opa builtin for gojq to query the entity object provided by the idp or the jwt with the subject_external_selector_value query string

references https://github.com/opentdf/platform/issues/470

### CODE REVIEW

**Review by pflynn-virtru — DISMISSED** [review]

**Comment by elizabethhealy** [comment]
> @elizabethhealy Should we be building a builtin for jq or should it be for subject mappings?

@strantalis the subject mapping evaluation takes place in the rego currently, this is just to retrieve the selected field in the jwt/entity obj per https://github.com/opentdf/platform/issues/470

File: service/internal/jqbuiltin/jq_builtin.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		if err != nil {
+			return nil, err
+		}
+
+		return ast.NewTerm(v), nil
+	},
+	)
+}
+
+func ExecuteQuery(inputJSON map[string]any, queryString string) ([]any, error) {
```
[the remark this answers is not in the document]
Comment (elizabethhealy) [reply service/internal/jqbuiltin/jq_builtin.go:51]: failsafe unescape added

**Review by jakedoublev — APPROVED** [review]
The use of JQ and the unit tests LGTM, but @pflynn-virtru or someone in Arch will need to review the authorization half of this for it to be merge-able. It feels a little odd that we need a plugin to support something the platform currently considers mandatory (subject condition sets with selector syntax and subject mappings), so maybe we should keep this in mind as we continue to evaluate the pros/cons of OPA for this evaluation.

**Comment by strantalis** [comment]
@elizabethhealy We want the subject mapping logic to also take place behind a builtin function as well. Someone that provides a custom opa bundle with modified rego shouldn't modify the mapping logic imo.

**Comment by elizabethhealy** [comment]
@strantalis ok that makes sense, i just read through the adr https://github.com/opentdf/platform/issues/472, we can open a separate issue when the adr is resolved to encompass that work (it will overwrite some of this work), but in the meantime can we merge this in to unblock downstream users since the other jq spec changes were already made?

**Comment by strantalis** [comment]
@ttschampel @jrschumacher @biscoe916 Can you give this a review.

**Comment by ttschampel** [comment]
I'm ok approving this to move forward and start work on #567 and get things aligned with #472

**Review by ttschampel — APPROVED** [review]
Lets quickly pick up https://github.com/opentdf/platform/issues/567 as a follow up

**Comment by elizabethhealy** [comment]
thanks @ttschampel, already getting started on the third bullet under #567

**Pull request merged**